File: patterns/javascript-patterns/pattern56.js
// Pattern 56: ...

// 54321
//  ****
//   321
//    **
//     1

const n = 5;

for (let i = n; i >= 1; i--) {
    let row = '';

    for (let j = n; j >= 1; j--) {
        if (i >= j) {
            if (i % 2 === 0) {
                row += '*';
            } else {
                row += j;
            }
        } else {
            row += ' ';
        }
    }

    console.log(row);
}

File: patterns/python-patterns/pattern56.py
# Pattern 56: ...

# 54321
#  ****
#   321
#    **
#     1

n = 5

for i in range(n, 0, -1):
    for j in range(n, 0, -1):
        if i >= j:
            if i % 2 == 0:
                print("*", end="")
            else:
                print(j, end="")
        else:
            print(" ", end="")
    print()
